store/src/java/com/zimbra/cs/mailbox/MailSender.java

```diff
@@ -1478,4 +1479,44 @@ public static void unregisterPreSendMailListener(PreSendMailListener listener) {
             }
         }
     }
+
+    /**
+     * provide the Hash for Message-Verification header field.
+     * @param id
+     * @param date
+     * @param from
+     * @throws ServiceException
+     */
+    public static String getHashForMessageVerification(String id, Date date, String from) throws MessagingException, ServiceException {
+        String guid = (id + date + from);
+        String guidHash = getEncryptedValue(guid, ALGORITHM_NAME);
+        String hash = HASH_ALGORITHM_INFO + guidHash;
+        return hash;
+    }
+
+    /**
+     * provide the Encrypted value of the GUID.
+     * @param input
+     * @param algo
+     * @throws ServiceException
+     */
+    public static String getEncryptedValue(String input, String algo) throws ServiceException {
+        try {
+            MessageDigest md1 = MessageDigest.getInstance(algo);
+            byte[] messageDigest1 = md1.digest(input.getBytes(StandardCharsets.UTF_8));
+            BigInteger number = new BigInteger(1, messageDigest1);
+            StringBuilder hexString = new StringBuilder(number.toString(16));
+
+            while (hexString.length() < 64) {
+                hexString.insert(0, '0');
+            }
+
+            String BasicBase64format
+                    = Base64.getEncoder()
+                    .encodeToString(hexString.toString().getBytes());
+            return BasicBase64format;
```

Comment:
Why is the `BasicBase64format` variable capitalized?

Reply:
Corrected , Thanks!